dotCMS/src/main/java/com/dotmarketing/business/BaseWebAssetAPI.java

```diff
@@ -298,17 +295,6 @@ else if(currWebAsset instanceof Template)
 				APILocator.getTemplateAPI().associateContainers(new ArrayList<Container>(), (Template)currWebAsset);
 				CacheLocator.getTemplateCache().remove(currWebAsset.getInode());
 			}
-			else if(currWebAsset instanceof Link)
-			{
-				VersionInfo vi = APILocator.getVersionableAPI().getVersionInfo(currWebAsset.getIdentifier());
-
-				if(!UtilMethods.isSet(vi)) {
-					auxVersionInfo = getVersionInfo(currWebAsset, identifier,
-							webAssetList, "links");
-				}
-
-				VirtualLinksCache.removePathFromCache(((Link)currWebAsset).getUrl());
-			}
 
 			if(auxVersionInfo==null || !UtilMethods.isSet(auxVersionInfo.getIdentifier())) { // null auxVersionInfo  indicates everything goes fine

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Change this condition so that it does not always evaluate to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2583)